### PR TITLE
Show interface name first

### PR DESCRIPTION
**Pull request by tyll**
When showing or editing a state, show the interface name first. This makes it easier to find a specific interface state or to edit the state in yaml, since the interface name now appears after the dash instead of other options that one might want to remove.

### CODE REVIEW

File: nmstatectl/nmstatectl.py [hunk 1]
```diff
@@ -93,10 +115,45 @@ def pretty_yaml(state):
...
+        ordered_state[key] = value
+
+    return ordered_state
+
+
+def order_state(state):
+    try:
+        iface_states = state.pop('interfaces')
+        state['interfaces'] = []
+        for iface_state in sorted(iface_states, key=itemgetter("name")):
```
[the remark this answers is not in the document]
Comment (tyll) [reply nmstatectl/nmstatectl.py]: changed

File: nmstatectl/nmstatectl.py [hunk 2]
```diff
@@ -93,10 +115,45 @@ def pretty_yaml(state):
...
+            iface_state = order_iface_state(iface_state)
+            state['interfaces'].append(iface_state)
+    except KeyError:
```
[the remark this answers is not in the document]
Comment (tyll) [reply nmstatectl/nmstatectl.py]: changed. Do not remember what it was needed for.

**Comment by tyll** [comment]
Everything should be addressed now. Not sure about travis - flake8 complains about the long URL but since it is a URL, breaking it into two lines would be bad, too.

**Comment by tyll** [comment]
I added type and state to the top as well

File: nmstatectl/prettystate.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    def json(self):
+        return json.dumps(self.state, indent=4, separators=(',', ': '))
+
+
+def represent_ordereddict(dumper, data):
+    """
+    Represent OrderedDict as regular dictionary
+
+    Source: https://stackoverflow.com/questions/16782112/can-pyyaml-dump-dict-items-in-non-alphabetical-order
+    """  # noqa: E501
```
Comment (EdDev) [inline nmstatectl/prettystate.py]: I prefer you break the line over breaking format rules.

File: nmstatectl/prettystate.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    return state
+
+
+def order_iface_state(iface_state):
+    ordered_state = OrderedDict()
+
+    for setting in ('name', 'type', 'state'):
+        try:
+            ordered_state[setting] = iface_state.pop(setting)
+        except KeyError:
```
Comment (EdDev) [inline nmstatectl/prettystate.py]: I guess you need this for the pop action.
How about this? (less noisy IMO):
```
key = iface_state.pop(setting, None)
if key is not None:
    ordered_state[setting] = key
```

File: nmstatectl/prettystate.py (hunk 3 above)
Comment (EdDev) [reply nmstatectl/prettystate.py]: 60 votes is not that convincing, but if you prefer it like this lets go with the exception.

File: nmstatectl/prettystate.py (hunk 4 above)
Comment (EdDev) [reply nmstatectl/prettystate.py]: The taste changes with context (and age :) )
Your argument is valid, leave it.

File: nmstatectl/prettystate.py (hunk 3 above)
Comment (tyll) [reply nmstatectl/prettystate.py]: All but one answer are for keeping the URL this way.

**Comment by tyll** [comment]
ci test please

**Pull request merged**